crates/bevy_ecs/src/world/mod.rs

```diff
@@ -627,8 +627,14 @@ impl World {
     }
 
     pub fn is_resource_added<T: Component>(&self) -> bool {
-        let component_id = self.components.get_resource_id(TypeId::of::<T>()).unwrap();
-        let column = self.get_populated_resource_column(component_id).unwrap();
+        let component_id = match self.components.get_resource_id(TypeId::of::<T>()) {
+            Some(id) => id,
+            None => return false,
+        };
+        let column = match self.get_populated_resource_column(component_id) {
+            Some(col) => col,
+            None => return false,
+        };
         // SAFE: resources table always have row 0
         let ticks = unsafe { column.get_ticks_unchecked(0) };
         ticks.is_added(self.last_change_tick(), self.read_change_tick())
```

Comment:
This *could* be combined with the line using [`Option::and_then`](https://doc.rust-lang.org/std/option/enum.Option.html#method.and_then), avoiding one match.

I would also change it to be an `if let Some(...)` with an `else {false}` at the end
I.e. my final version would be:
```suggestion
        if let Some(column) = self.components.get_resource_id(TypeId::of::<T>())
             .and_then(|component_id| self.get_populated_resource_column(component_id)) {
            // SAFE: resources table always have row 0
            let ticks = unsafe { column.get_ticks_unchecked(0) };
            ticks.is_added(self.last_change_tick(), self.read_change_tick())
        } else {
            false
        }
```

Reply:
it seems a bit less readable imo, the if condition stretches over 2 lines(or occupies a really long line) and generally seems a bit crowded, the match cases(while seems a bit long) gives you a clear view of what is going on at a glance.
(and generally speaking i would get that if condition to be a variable and a short `if let` statement instead)